src/coprocessor/dag/aggr_fn/impl_max_min.rs

```diff
@@ -0,0 +1,342 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::Ordering;
+use std::convert::TryFrom;
+
+use cop_codegen::AggrFunction;
+use cop_datatype::{EvalType, FieldTypeAccessor};
+use tipb::expression::{Expr, ExprType, FieldType};
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::mysql::Tz;
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::dag::rpn_expr::{RpnExpression, RpnExpressionBuilder};
+use crate::coprocessor::Result;
+
+pub trait Extremum {
+    const TP: ExprType;
+    const ORD: Ordering;
+}
+
+macro_rules! extremum {
+    ($e:ident, $ord:path) => {
+        pub struct $e;
+        impl Extremum for $e {
+            const TP: ExprType = ExprType::$e;
```

Comment:
This is pretty tricky. It won't work in future. Let's simply remove this macro, since it is very small and we are unlikely to have more in future as well.

Reply:
OK, It's just a style issue.